### PR TITLE
Api 24176/spike export oauth tool as a package

### PR DESCRIPTION
**Pull request by stevelong00**
This is a branch with working code that refactors and wraps 3 of the 4 CLI tools as a node package. The 4th CLI tool is a bash script and cannot be wrapped.

An example project showing how to consume the node package has been added under the folder **npm-example**.  For now, this consists primarily of a README and some simple code that I have been using to run manual tests.  This projects expects the auth-utils to have been packaged into a zip file, so before running **npm install** on the example project be sure to do this packaging.  Packaging can be accomplished by running **npm run build && npm run deploy** in the root of the repo.

Things to do in follow up tickets:

- Rename the following files.  The naming convention currently in use was chose to make diffs easier to follow.  Once these proposed changes are accepted this convention is preferred.

```
auth.js -> auth-lib.js
auth-script.js -> auth.js
auth-cc.js -> auth-cc-lib.js
auth-cc-script.js -> aith-cc.js
client-assertion.js -> client-assertion-lib.js
client-assertion-script.js -> client-assertion.js   
```
- In package.json, change the deploy script to publish to a registry instead of packaging a zip file locally.  Update the example project to consume this.
- In package.json, connect this version number to the CI/CD tools so it is updated automatically.

### CODE REVIEW

File: auth-cc-script.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env node
+
+import { setOptions, getMetadata, getAssertion, getToken, argv } from "./auth-cc.js";
+import process from "process";
+
```
Comment (mheusgit) [inline auth-cc-script.js:5]: A comment about how this is used would be nice

File: auth-script.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env node
+
+import { setOptions, getCode, getTokens, AuthError, argv } from "./auth.js";
+import process from "process";
+
```
Comment (mheusgit) [inline auth-script.js:5]: A comment how this is different than auth-cc-script.js would be helpful

**Review by crolarlibertyva — CHANGES_REQUESTED** [review]
Are the script renames necessary? These will break regression tests that we have in place for oauth-proxy as well as some of our monitoring bots.

**Comment by christineyost** [comment]
Hi @crolarlibertyva this is a draft PR to show the types of changes we plan to make to be able to import these utilities as a module into Docserver. We will not be merging these changes in as they are, and we will not change the name of existing files in our final changes. We don't want to break any consumers of this tool! The file names we intend to use are detailed in the top comment

**Comment by dbrown-va** [comment]
Closing as out of date in favor of the changes in https://github.com/department-of-veterans-affairs/lighthouse-auth-utils/pull/57

**Pull request closed without merging**